### DISCUSSION TITLE
Reset form but set all fields as dirty

### DISCUSSION THREAD

**Discussion by mzaharie** [react-hook-form/react-hook-form#2999] [Q&A] [answered]
The use case is called Duplicate. This means the user can go from `products/:id`(form is not dirty here) to `products/create` and have the  form reset then every single input field being dirty with data in it.

How can I easily accomplish this  with RHF? There are a lot of fields and fieldArrays in the form!

**Comment by bluebill1049** [comment]
make all the input dirty? hmmmm that's tricky, or you can force that with `setValues`? but that's the thing dirty is for checking if the values are changed.

**Comment by mzaharie** [comment, marked as answer]
The easy workaround was to reset one of the inputs to a custom value that is hard to guess by the user. Then use `setValue(field, correctValue, { shouldDirty: true })`.
